jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpanContext.java

```diff
@@ -135,7 +135,7 @@ public boolean isSampled() {
   }
 
   public boolean isDebug() {
-    return (flags & flagDebug) == flagDebug;
+    return (flags & flagDebug) == flagDebug || debugId != null;
```

Comment:
Why this check? Can we keep this class without changes?

Reply:
isDebug is decided purely on the flag, whether there is a correlation identifier is irrelevant.